src/main/java/edu/cornell/kfs/ksr/document/validation/impl/SecurityRequestDocumentRule.java

```diff
@@ -56,8 +57,8 @@ protected Map<String,String[]> getRequiredQualificationsMap() {
 
     @Override
     public boolean isDocumentAttributesValid(Document document, boolean validateRequired) {
-        boolean success = true;
         SecurityRequestDocument securityRequestDocument = (SecurityRequestDocument) document;
+        boolean success = validateDepartmentCode(securityRequestDocument.getPrimaryDepartmentCode());

```

Comment:
Regex-based validation of properties can be configured at the Data Dictionary level, and it looks like such a configuration is already present on the "primaryDepartmentCode" property. I would suggest using just the DD-based validation instead, if possible.

It looks like the "primaryDepartmentCode" DD config is specifying the "validCharactersConstraint" instead of the "validationPattern". Several other KSR beans also have such a config; this must have been part of the KNS-to-KRAD conversion back on the Cynergy side, which we now have to undo. Fixing this DD config accordingly might be enough to fix the problem.